### PR TITLE
Removing a dangerous fscanf usage

### PR DESCRIPTION
**Pull request by raulgarciamsft**
Fixing a potential BO that could happen if the ID assigned by the trustee would exceed the limit in the ballot files and adding a test for the fix.

### Issue
*Link your PR to an issue*

Fixes #___

### Description
*Please describe your pull request.*

### Testing
I added a test that tries to read pre-populated ballots with valid & invalid external identifiers.

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] ✅ **DO** check open PR's to avoid duplicates.
- [x] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [x] ✅ **DO** build locally before pushing.
- [x] ✅ **DO** make sure tests pass.
- [x] ✅ **DO** make sure any new changes are documented.
- [x] ✅ **DO** make sure not to introduce any compiler warnings.
- [x] ❌**AVOID** breaking the continuous integration build.
- [x] ❌**AVOID** making significant changes to the overall architecture.

💔Thank you!

### CODE REVIEW

**Comment by raulgarciamsft** [comment]
Any updates on this one?

**Comment by raulgarciamsft** [comment]
Any update on this PR? Thanks

File: Makefile.mk [hunk 1]
```diff
@@ -62,6 +62,18 @@ else
...
+run-ballot-parser: build
+	if [ ! -d "ballot_parser_build" ]; then mkdir ballot_parser_build; fi
+ifeq ($(OPERATING_SYSTEM),Windows)
+	CMAKE_PREFIX_PATH="./build/ElectionGuard" cmake -S examples/ballot_parser -B ballot_parser_build -G "MSYS Makefiles"
+	cmake --build ballot_parser_build --target ballot_parser
+	PATH=$(PWD)/build:$$PATH; ./ballot_parser_build/ballot_parser
+else
+	ElectionGuard_DIR=$(ELECTIONGUARD_DIR) cmake -S examples/ballot_parser -B ballot_parser_build
+	cmake --build ballot_parser_build --target ballot_parser
+	./ballot_parser_build/ballot_parser
```
Comment (AddressXception) [inline Makefile.mk]: maybe pass in the default `./examples/ballot_parser/ballot_samples` path here?

**Review by AddressXception — CHANGES_REQUESTED** [review]
lg2m.  Just one comment about passing in the default ballots into the makefile so the `run-ballot-parser` command executes

File: Makefile.mk (hunk 1 above)
Comment (raulgarciamsft) [reply Makefile.mk]: Fixed

**Comment by raulgarciamsft** [comment]
I am hitting a code unrelated issue trying to run the Linux build (updating cmake)

**Review by AddressXception — APPROVED** [review]

**Pull request merged**